1.sh

```diff
@@ -1,35 +1,34 @@
 #! /bin/sh
 
-if [-d $1] && [-d $2]
+if [ -d "$1" ] && [ -d "$2" ]
 then
 	echo "Dirs exist"
-elif [! -d $2] 
+elif [ ! -d "$2" ] 
 	echo "Flash path doesn't exist"
-	mkdir $2
+	mkdir "$2"
 fi
 
-for file in $1/*
+for file in "$1"/*
 do
-	echo $(basename $file)
-	flashfile=$($2/$(basename $file))
-	if [! -f $flashfile]; then
-		echo "Copying files to $2"
-		cp $file $2
+	echo "$(basename $file)"
+	flashfile="$2/$(basename $file)"
+
+	if [ ! -f "$flashfile" ]; then
+		echo "Copying files to '$2'"
+		cp "$file" "$2"
 	
 	elif [ $(ls -l $file) qt $(ls -l $flashfile) ]
-		echo "Flash version is newer, replace existing one? Y\N"
-
-	read reply
-	case $reply in
-	y|Y) echo "Copying now..."
-		cp $flashdrive $1
-	;;
-	n|N) echo "Staying as we were" 
-		exit 2
-	*) echo "Wrong input"
-	;;
-	esac
+		read -e -n 1 -p "Flash version is newer, replace existing one? [Y|N] "
+		case "$REPLY" in
+			y|Y)
+				echo "Copying now..."
+				cp "$flashdrive" "$1" ;;
+			n|N)
+				echo "Staying as we were" 
+				exit 2 ;;
+			*)
+				echo "Wrong input" ;;
```

Comment:
Remember, a `;;` must be given after the code here in a case statement, as shown above. It can either be on the end of the last line of command(s) or alone on one of the following lines.